### DISCUSSION TITLE
Flash using the builder image

### DISCUSSION THREAD

**Discussion by dhamidou** [bitcraze/discussions#1209] [Q&A] [answered]
Hi, I attempt to follow this [https://github.com/bitcraze/aideck-esp-firmware](https://github.com/bitcraze/aideck-esp-firmware), to flash using the builder image (I use a jlink for that) but I have this error : 
![Capture d’écran de 2024-03-01 14-51-57](https://github.com/bitcraze/discussions/assets/95414130/835d95c8-4a55-43aa-bf75-3411adea317b)
Can you help me ? 
Thank you in advance

**Comment by gemenerik** [comment]
I'll verify I can flash the ESP with a j-link, but as a workaround you can always just flash the ESP with [a firmware update through cfclient](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#update-crazyflie-and-aideck-firmware).

**Reply by gemenerik** [reply]
I run into the same problem, but managed to flash it by lowering the adapter khz.

```
docker run --rm -it -v $PWD:/module/ --device /dev/ttyUSB0 --privileged -P bitcraze/builder /bin/bash -c "openocd -f interface/jlink.cfg -f board/esp-wroom-32.cfg -c 'adapter_khz 14999' -c 'program_esp build/bootloader/bootloader.bin 0x1000 verify' -c 'program_esp build/aideck_esp.bin 0x10000 verify reset exit'"
```

**Reply by gemenerik** [reply]
> but managed to flash it by lowering the adapter khz

It appears to break at 15000 Hz

**Reply by dhamidou** [reply]
I tried the same command as you but it displays the same thing. Additionally, when I flash the ESP with a firmware update through cfclient, I don't have the "ESP32: I (910) SYS: Initialized" step.

**Reply by gemenerik** [reply]
Can you lower the adapter Hz further? E.g., try 10000
Can you verify that you correctly connected your JTAG debugger?

**Reply by dhamidou** [reply]
Hi, sorry for taking a while to respond. I even lowered it to 1KHz but nothing happened. And the jlink is connected well

**Reply by gemenerik** [reply]
No worries! Can you verify that during the firmware update through the client you had no other decks connected, and were powered on battery? The `ESP32: I (910) SYS: Initialized` should show up in the cfclient console after connecting.

**Reply by dhamidou** [reply]
through the client (no other deck and powered battery):
![Capture d’écran de 2024-03-15 10-51-06](https://github.com/bitcraze/discussions/assets/95414130/ee81ecb8-ac5c-4796-8151-b4265f95e162)
through the jlink :
![Capture d’écran de 2024-03-15 10-53-23](https://github.com/bitcraze/discussions/assets/95414130/d8d94dc5-44e4-4665-892e-76815c0d4c6a)

**Reply by gemenerik** [reply, marked as answer]
I do actually see the `ESP32: SYS: Initialized` message in your client console, so it appears that through the firmware update, the ESP32 has been flashed successfully. That implies you don't have to bother with flashing the ESP with your J-Link.

If you want to be able to flash the ESP32 with your J-Link anyhow, you can perhaps try debugging some other chip. E.g., try flashing the [GAP8 bootloader](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#gap8-bootloader) (if so make sure you connect to the GAP8 instead). That can help narrow down the problem.

**Reply by dhamidou** [reply]
ok thank you
